src/Microsoft.Extensions.Configuration/ChainedConfigurationProvider.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public class ChainedConfigurationProvider : ConfigurationProvider
```

Comment:
Once we decide the best name for the Add extension method, try to make the name of the class align (or the other way around).